crates/build/src/manifest.rs

```diff
@@ -8,6 +8,31 @@ use spin_manifest::{schema::v2, ManifestVersion};
 /// given (v1 or v2) manifest path.
 pub async fn component_build_configs(
```

Comment:
Perhaps rename this `component_build_configs_with_error` or something? And update the docstring?

Reply:
I'm not mad keen on that name (and can't think of a fully expressive one that isn't awkward) but have updated the doc string.